src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -127,6 +159,34 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
     this
   }
 
+  /*
+   * Implemented by means of a buffer to keep the last n elements from being returned during iteration.
+   */
+  def dropRight(n: Int): Iterator[A] = {
```

Comment:
Should we have this kind of operation that requires buffering at all on `Iterator`? And if so, what is the expectation when the buffering happens? Upon construction of the `Iterator` as implemented here, or lazily on the first access?